lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -369,6 +369,18 @@ func (c *Client) start() {
 				} else {
 					c.cfg.Log.Warning("Recieved an empty clipboard message")
 				}
+			case tdp.SharedDirectoryAnnounce:
+				if c.cfg.AllowDirectorySharing {
+					driveName := C.CString(m.Name)
+					defer C.free(unsafe.Pointer(driveName))
+					if err := C.handle_tdp_sd_announce(c.rustClient, C.CGOSharedDirectoryAnnounce{
+						directory_id: C.uint32_t(m.DirectoryID),
+						name:         driveName,
+					}); err != C.ErrCodeSuccess {
+						c.cfg.Log.Errorf("Device announce failed: %v", err)
+						return
```

Comment:
This is in keeping with how we handle the other TDP --> RDP translation failures. 

This could be improved by sending back a `SharedDirectoryAcknowledge` with a failure code in cases where handle_tdp_sd_announce fails, and only returning a `CGOErrCode::ErrCodeFailure` when an irrecoverable failure occurs like https://github.com/gravitational/teleport/blob/332af8563418474fa93401b12746fbf5cbd9c552/lib/srv/desktop/rdp/rdpclient/src/lib.rs#L485-L490

Reply:
> Question is, how likely is it that we see an error here? If not likely, then we can stick with the existing pattern.

Based on my experience working with this thus far, I think it's unlikely enough that we are fine to leave this as is for now.

I agree with all of that. I had the same general thought while developing this feature, noticing that we didn't have a consistent system for when we kill the connection (typically by returning an `Err(...)` for an `RdpResult`, vs when we send back an RDP response with `NTSTATUS::STATUS_UNSUCCESSFUL`, vs when we send a TDP error between those options.

I created an issue to remind us to go back and do that: https://github.com/gravitational/teleport/issues/13041